components/formats-bsd/test/loci/formats/utests/FileStitcherTest.java

```diff
@@ -193,4 +212,43 @@ public void testStitch(Integer[] dims) throws IOException, FormatException {
     check(pattern, filenames.toArray(new String[filenames.size()]), dims);
   }
 
+  @Test
+  public void testUnderlyingReaders() throws IOException, FormatException {
+    FakeReader reader = new FakeReader();
+    FileStitcher fs = new FileStitcher(reader);
+    assertNotNull(fs.getUnderlyingReaders());
+    fs.setId("test_z<0-2>.fake");
+    assertNotNull(fs.getUnderlyingReaders());
```

Comment:
As discussed above, the `getUnderlyingReaders` protocol is not well defined, and in `FileStitcher` it also depends on the state, so I didn't feel like adding a more specific test here. In a way, unit tests are part of the docs. I can add a length check before and after `setId` if you want, but I think this would make the current behaviour more "official".